### PR TITLE
Update version for junit platform and fix integraiton tests

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

- Bugfix

### Description

Fix for https://github.com/strimzi/strimzi-kafka-operator/issues/2115#event-2729165663

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests

**Comment by tombentley** [comment]
AFAIC we currently have both junit 4 and 5 as dependencies in the api. Can anyone explain why that is?

**Comment by Frawless** [comment]
@tombentley that because integration tests use the same base as system tests. There is several ways how to solve that:
1. Create different base with mostly same code for integration tests and write them back to junit4 (junit5 will be only in systemtest package after that)
2. Migrate unit tests to junit5 and remove junit4 dependency from whole project (I prefer this one)

**Comment by tombentley** [comment]
@Frawless I agree that having a single version in the project would be an improvement. But we need to understand what support vertx has for 5, since we use their runner for a lot of the tests. https://vertx.io/docs/vertx-junit5/java/

**Comment by Frawless** [comment]
@tombentley we did migration for bridge tests where Paolo used basically same thing - vertx runner can be replaced with vertx extension, EnMasse migrated all unit tests to junit5 like 1 year ago and I think they had same pattern for the tests

**Comment by ppatierno** [comment]
@Frawless I still don't see the CrdIT tests running on Travis

**Comment by Frawless** [comment]
@ppatierno I found this in travis log
```make[1]: Leaving directory '/home/travis/build/strimzi/strimzi-kafka-operator/crd-generator'
make -C api docker_build
make[1]: Entering directory '/home/travis/build/strimzi/strimzi-kafka-operator/api'
echo "Installing root pom ..."
Installing root pom ...
mvn -q install -f ../pom.xml -N
echo "Installing JAR files ..."
Installing JAR files ...
mvn -q install
2019-10-21 13:07:43 INFO [BaseITST:166] Creating namespace: kafkamirrormaker-crd-it
2019-10-21 13:07:44 INFO [BaseITST:172] Using namespace kafkamirrormaker-crd-it
2019-10-21 13:07:44 INFO [BaseITST:64] Changing to kafkamirrormaker-crd-it namespace
2019-10-21 13:07:44 INFO [BaseITST:207] Creating resources ../install/cluster-operator/045-Crd-kafkamirrormaker.yaml
2019-10-21 13:07:52 INFO [BaseITST:219] Deleting resources ../install/cluster-operator/045-Crd-kafkamirrormaker.yaml
2019-10-21 13:07:53 INFO [BaseITST:191] Deleting namespace: kafkamirrormaker-crd-it
2019-10-21 13:07:58 INFO [BaseITST:196] Using namespace default
2019-10-21 13:07:58 INFO [BaseITST:64] Changing to default namespace
2019-10-21 13:07:58 INFO [BaseITST:166] Creating namespace: kafkacrd-it
2019-10-21 13:07:58 INFO [BaseITST:172] Using namespace kafkacrd-it
2019-10-21 13:07:58 INFO [BaseITST:64] Changing to kafkacrd-it namespace
2019-10-21 13:07:58 INFO [BaseITST:207] Creating resources ../install/cluster-operator/040-Crd-kafka.yaml
2019-10-21 13:08:10 INFO [BaseITST:219] Deleting resources ../install/cluster-operator/040-Crd-kafka.yaml
2019-10-21 13:08:11 INFO [BaseITST:191] Deleting namespace: kafkacrd-it
2019-10-21 13:08:17 INFO [BaseITST:196] Using namespace default
2019-10-21 13:08:17 INFO [BaseITST:64] Changing to default namespace
2019-10-21 13:08:17 INFO [BaseITST:166] Creating namespace: kafkaconnect-crd-it
2019-10-21 13:08:17 INFO [BaseITST:172] Using namespace kafkaconnect-crd-it
2019-10-21 13:08:17 INFO [BaseITST:64] Changing to kafkaconnect-crd-it namespace
2019-10-21 13:08:17 INFO [BaseITST:207] Creating resources ../install/cluster-operator/041-Crd-kafkaconnect.yaml
2019-10-21 13:08:26 INFO [BaseITST:219] Deleting resources ../install/cluster-operator/041-Crd-kafkaconnect.yaml
2019-10-21 13:08:26 INFO [BaseITST:191] Deleting namespace: kafkaconnect-crd-it
2019-10-21 13:08:32 INFO [BaseITST:196] Using namespace default
2019-10-21 13:08:32 INFO [BaseITST:64] Changing to default namespace
2019-10-21 13:08:32 INFO [BaseITST:166] Creating namespace: topiccrd-it
2019-10-21 13:08:32 INFO [BaseITST:172] Using namespace topiccrd-it
2019-10-21 13:08:32 INFO [BaseITST:64] Changing to topiccrd-it namespace
2019-10-21 13:08:32 INFO [BaseITST:207] Creating resources ../install/cluster-operator/043-Crd-kafkatopic.yaml
2019-10-21 13:08:37 INFO [BaseITST:219] Deleting resources ../install/cluster-operator/043-Crd-kafkatopic.yaml
2019-10-21 13:08:37 INFO [BaseITST:191] Deleting namespace: topiccrd-it
2019-10-21 13:08:43 INFO [BaseITST:196] Using namespace default
2019-10-21 13:08:43 INFO [BaseITST:64] Changing to default namespace
2019-10-21 13:08:43 INFO [BaseITST:166] Creating namespace: kafkausercrd-it
2019-10-21 13:08:43 INFO [BaseITST:172] Using namespace kafkausercrd-it
2019-10-21 13:08:43 INFO [BaseITST:64] Changing to kafkausercrd-it namespace
2019-10-21 13:08:43 INFO [BaseITST:207] Creating resources ../install/cluster-operator/044-Crd-kafkauser.yaml
2019-10-21 13:08:47 INFO [BaseITST:219] Deleting resources ../install/cluster-operator/044-Crd-kafkauser.yaml
2019-10-21 13:08:47 INFO [BaseITST:191] Deleting namespace: kafkausercrd-it
2019-10-21 13:08:53 INFO [BaseITST:196] Using namespace default
2019-10-21 13:08:53 INFO [BaseITST:64] Changing to default namespace
2019-10-21 13:08:53 INFO [BaseITST:166] Creating namespace: kafkabridge-crd-it
2019-10-21 13:08:53 INFO [BaseITST:172] Using namespace kafkabridge-crd-it
2019-10-21 13:08:53 INFO [BaseITST:64] Changing to kafkabridge-crd-it namespace
2019-10-21 13:08:53 INFO [BaseITST:207] Creating resources ../install/cluster-operator/046-Crd-kafkabridge.yaml
2019-10-21 13:09:00 INFO [BaseITST:219] Deleting resources ../install/cluster-operator/046-Crd-kafkabridge.yaml
2019-10-21 13:09:00 INFO [BaseITST:191] Deleting namespace: kafkabridge-crd-it
2019-10-21 13:09:06 INFO [BaseITST:196] Using namespace default
2019-10-21 13:09:06 INFO [BaseITST:64] Changing to default namespace```

which corresponds with IT logs. Maven is executed with `-q` cause travis. So I considered that as success with fix of https://github.com/strimzi/strimzi-kafka-operator/issues/2115

File: api/src/test/resources/io/strimzi/api/kafka/model/Kafka.out.yaml [hunk 1]
```diff
@@ -25,6 +23,8 @@ spec:
         - podSelector:
             matchLabels:
               app: "kafka-producer"
+        authentication:
+          type: "scram-sha-512"
```
Comment (ppatierno) [inline api/src/test/resources/io/strimzi/api/kafka/model/Kafka.out.yaml]: why moving authentication section here?

Comment (Frawless) [reply api/src/test/resources/io/strimzi/api/kafka/model/Kafka.out.yaml]: Test which use this as expected output failed every time before move this part, problem in output verification. Not sure why it happen only in this PR and not early.

**Pull request merged**